app/scripts/services/bindingModalUtils.js

```diff
@@ -0,0 +1,24 @@
+'use strict';
+
+angular.module('openshiftConsole')
+  .factory('BindingModalUtils', [
```

Comment:
Instead of making it specific to the binding modal, this could be a general utility that combines and (optionally?) sorts collections of API objects.

We have this need for config maps and secrets in some places where we combine them in a select.